src/pip/_internal/req/req_file.py

```diff
@@ -23,7 +23,7 @@
 __all__ = ['parse_requirements']
 
 SCHEME_RE = re.compile(r'^(http|https|file):', re.I)
-COMMENT_RE = re.compile(r'(^|\s)+#.*$')
+COMMENT_RE = re.compile(r'\s*#.*$')
```

Comment:
What about:
```
COMMENT_RE = re.compile(r'(^|\s+)#.*$')
```
?

Reply:
testing

Reply:
Hmm, this still fails on `tests/functional/test_install_reqs.py`

Reply:
Really? The whole testsuite passes locally. What are the failing tests? Which Python version?

Reply:
How do you run the testsuite?